Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -199,6 +197,28 @@ private Uri GetAppIdEndPointUri(string instrumentationKey)
             return new Uri(this.endpointAddress, string.Format(CultureInfo.InvariantCulture, AppIdQueryApiRelativeUriFormat, instrumentationKey));
         }
 
+        /// <summary>
+        /// Registers failure for further action in future.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which the failure occurred.</param>
+        /// <param name="ex">Exception indicating failure.</param>
+        private void RegisterFailure(string instrumentationKey, Exception ex)
+        {
+            var webException = ex as WebException;
+            if (webException != null)
+            {
+                this.failingInstrumenationKeys[instrumentationKey] = new FailedResult(
+                    DateTime.UtcNow,
+                    ((HttpWebResponse)webException.Response).StatusCode);
+            }
+            else
+            {
+                this.failingInstrumenationKeys[instrumentationKey] = new FailedResult(DateTime.UtcNow);
```

Comment:
Suggestion:  Add a ```FailedResult``` constructor that takes an exception and move this logic there. This method then becomes a one-liner and you can remove the other unused constructors.
```C#
this.failingInstrumentationKeys[instrumentationKey] = new FailedResult(DateTime.UtcNow, ex);